src/Symfony/Bridge/PhpUnit/Tests/EnumExistsMockTest.php

```diff
@@ -0,0 +1,80 @@
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\PhpUnit\Tests;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Bridge\PhpUnit\ClassExistsMock;
+
+/**
+ * @requires PHP 8.1
+ */
+class EnumExistsMockTest extends TestCase
+{
+    public static function setUpBeforeClass(): void
+    {
+        ClassExistsMock::register(__CLASS__);
+    }
+
+    protected function setUp(): void
+    {
+        ClassExistsMock::withMockedEnums([
+            ExistingEnum::class => false,
+            'NonExistingEnum' => true,
+        ]);
+    }
+
+    public function testClassExists()
+    {
+        $this->assertFalse(class_exists(ExistingEnum::class));
+        $this->assertFalse(class_exists(ExistingEnum::class, false));
+        $this->assertFalse(class_exists('\\'.ExistingEnum::class));
+        $this->assertFalse(class_exists('\\'.ExistingEnum::class, false));
+        $this->assertTrue(class_exists('NonExistingEnum'));
+        $this->assertTrue(class_exists('NonExistingEnum', false));
+        $this->assertTrue(class_exists('\\NonExistingEnum'));
+        $this->assertTrue(class_exists('\\NonExistingEnum', false));
+        $this->assertTrue(class_exists(ExistingEnumReal::class));
+        $this->assertTrue(class_exists(ExistingEnumReal::class, false));
+        $this->assertTrue(class_exists('\\'.ExistingEnumReal::class));
+        $this->assertTrue(class_exists('\\'.ExistingEnumReal::class, false));
+        $this->assertFalse(class_exists('\\NonExistingEnumReal'));
+        $this->assertFalse(class_exists('\\NonExistingEnumReal', false));
+    }
+
+    public function testEnumExists()
+    {
+        $this->assertFalse(enum_exists(ExistingEnum::class));
+        $this->assertFalse(enum_exists(ExistingEnum::class, false));
+        $this->assertFalse(enum_exists('\\'.ExistingEnum::class));
+        $this->assertFalse(enum_exists('\\'.ExistingEnum::class, false));
+        $this->assertTrue(enum_exists('NonExistingEnum'));
+        $this->assertTrue(enum_exists('NonExistingEnum', false));
+        $this->assertTrue(enum_exists('\\NonExistingEnum'));
+        $this->assertTrue(enum_exists('\\NonExistingEnum', false));
+        $this->assertTrue(enum_exists(ExistingEnumReal::class));
+        $this->assertTrue(enum_exists(ExistingEnumReal::class, false));
+        $this->assertTrue(enum_exists('\\'.ExistingEnumReal::class));
+        $this->assertTrue(enum_exists('\\'.ExistingEnumReal::class, false));
+        $this->assertFalse(enum_exists('NonExistingClassReal'));
+        $this->assertFalse(enum_exists('NonExistingClassReal', false));
+        $this->assertFalse(enum_exists('\\NonExistingEnumReal'));
+        $this->assertFalse(enum_exists('\\NonExistingEnumReal', false));
+    }
+}
+
+enum ExistingEnum
+{
+}
+
+enum ExistingEnumReal
+{
+}
```

Comment:
This will cause a parse error on PHP < 8.1. If we want to be able to run the tests on lower PHP versions, those two enums have to be moved into separate files.

Reply:
Fixed, thanks!